mgmtsystem_action/models/mgmtsystem_action.py

```diff
@@ -0,0 +1,185 @@
+        action = super(MgmtSystemAction, self).create(vals)
+        template.send_mail(action.id, force_send=True)
+        return action
+
+    @api.multi
+    def write(self, vals):
+        """Update user data."""
+        stage_open = self.env.ref('mgmtsystem_action.stage_open')
+        stage_close = self.env.ref('mgmtsystem_action.stage_close')
+        if vals.get('stage_id'):
+            if vals['stage_id'] == stage_open.id:
+                vals['opening_date'] = datetime.now()
+            if vals['stage_id'] == stage_close.id:
+                vals['date_closed'] = datetime.now()
+
+        result = super(MgmtSystemAction, self).write(vals)
+        return result
+
+    def send_mail_for_action(self, action):
+        """Set a document state as draft and notified the reviewers."""
+        template = self.env.ref(
+            'mgmtsystem_action.email_template_new_action_reminder')
+
+        template.send_mail(action.id, force_send=True)
+        return True
+
+#    @api.multi
+#    def message_auto_subscribe(self, updated_fields, values=None):
+#        """Automatically add the responsible user to the follow list."""
+#        for o in self:
+#            self.message_subscribe_users(user_ids=[o.user_id.id],
+#                                         subtype_ids=None)
+#
+#        base = super(MgmtSystemAction, self)
+#        return base.message_auto_subscribe(updated_fields, values=values)
+
+    @api.multi
+    def case_open(self):
+        """Open case."""
+        for case in self:
+            values = {'active': True}
+
+            stages = self.env['mgmtsystem.action.stage']
+            values['stage_id'] = stages.search([
+                ['is_ending', '=', False],
+                ['is_starting', '=', False]
+            ]).id
+
+            case.write(values)
+
+        return True
+
+    def get_action_url(self):
+        """Return action url."""
+        base_url = self.env['ir.config_parameter'].get_param(
+            'web.base.url',
+            default='http://localhost:8069'
+        )
+        url = ('{}/web#db={}&id={}&model={}').format(
+            base_url,
+            self.env.cr.dbname,
+            self.id,
+            self._name
+        )
+        return url
+
+    @api.model
+    def _stage_groups(self, present_ids, domain, **kwargs):
+        """This method is used by KanBan view to show empty stages."""
+        stage_obj = self.env['mgmtsystem.action.stage']
+
+        # perform search
+        stage_ids = stage_obj._search([])
+        result = stage_obj.search([]).name_get()
+        # restore order of the search
+        result.sort(lambda x, y: cmp(
+            stage_ids.index(x[0]), stage_ids.index(y[0])))
+
+        fold = {}
+        for stage in stage_obj.browse(stage_ids):
+            fold[stage.id] = True
+        return result, None
+
+    @api.model
+    def process_reminder_queue(self):
+        """Notify user when we are 10 days close to a deadline."""
```

Comment:
I feel that this could be replaces by an Automated Action.
It's OK for me if you just add the idea to the Roadmap.